source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,309 @@
+===================
+Compound Operations
+===================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to perform **compound operations**.
+
+Compound operations combine a read and write operation into
+a single operation.  If you perform a read and write operation
+separately, there's a chance someone else may alter the document between
+both operations. MongoDB prevents this by placing a write lock on the
+document you are modifying for the duration of your compound operation.
+
+MongoDB supports the following compound operations:
+
+- :ref:`Find and delete one document <go-find-and-delete>`
+- :ref:`Find and update one document <go-find-and-update>`
+- :ref:`Find and replace one document <go-find-and-replace>`
+
+.. tip::
+    
+   If you need to read and write to more than one document, use
+   :manual:`transactions </core/transactions/>`.
+
+Sample Data
+~~~~~~~~~~~
+
+Run the following snippet to load the documents into the ``ratings``
+collection of the ``tea`` database:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/compoundOperations.go
+   :language: go
+   :dedent:
+   :start-after: begin insertDocs
+   :end-before: end insertDocs
+
+.. include:: /includes/fundamentals/tea-sample-data-ending.rst
+
+.. _go-find-and-delete:
+
+Find and Delete
+---------------
+
+To find and delete one document, pass a ``Context`` type and a
+``non-nil`` query filter specifying which documents to match to the
+``FindOneAndDelete()`` function. 
```

Comment:
More description about what `FindOneAndDelete` does would be helpful here before jumping into the function parameters. I had to read the driver docs to figure out how `FindOneAndDelete` was different than `DeleteOne` (returns the deleted doc instead of just deleting it), which I suspect other readers might not understand at first either.